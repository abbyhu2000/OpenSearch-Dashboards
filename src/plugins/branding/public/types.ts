import { NavigationPublicPluginStart } from '../../navigation/public';

export interface BrandingPluginSetup {
  getGreeting: () => string;
}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface BrandingPluginStart {}

export interface AppPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
}
