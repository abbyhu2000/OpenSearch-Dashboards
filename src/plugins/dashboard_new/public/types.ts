/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { SavedObject } from '../../saved_objects/public';
import { Filter, ISearchSource, Query, RefreshInterval } from '../../data/public';
import { SavedDashboardPanel730ToLatest } from '../common';
import { ViewMode } from 'src/plugins/embeddable/public';

/**
 * This should always represent the latest dashboard panel shape, after all possible migrations.
 */
export type SavedDashboardPanel = SavedDashboardPanel730ToLatest;

export interface SavedDashboardState {
  panels: SavedDashboardPanel[];
  fullScreenMode: boolean;
  title: string;
  description: string;
  timeRestore: boolean;
  options: {
    hidePanelTitles: boolean;
    useMargins: boolean;
  };
  viewMode: ViewMode;
  expandedPanelId?: string;
}

export interface ISavedDashboard {
  id?: string;
  timeRestore: boolean;
  timeTo?: string;
  timeFrom?: string;
  description?: string;
  panelsJSON: string;
  optionsJSON?: string;
  // TODO: write a migration to rid of this, it's only around for bwc.
  uiStateJSON?: string;
  lastSavedTitle: string;
  refreshInterval?: RefreshInterval;
  searchSource?: ISearchSource;
  getQuery(): Query;
  getFilters(): Filter[];
}

export interface DashboardSavedObject extends SavedObject, ISavedDashboard {}

interface FieldSubType {
  multi?: { parent: string };
  nested?: { path: string };
}

export interface Field {
  name: string;
  type: string;
  // esTypes might be undefined on old index patterns that have not been refreshed since we added
  // this prop. It is also undefined on scripted fields.
  esTypes?: string[];
  aggregatable: boolean;
  filterable: boolean;
  searchable: boolean;
  subType?: FieldSubType;
}

export type NavAction = (anchorElement?: any) => void;

// TODO: this might not be needed because we have setDashboard(partial) now
// /**
//  * In URL panels are optional,
//  * Panels are not added to the URL when in "view" mode
//  */
// export type DashboardAppStateInUrl = Omit<DashboardAppState, 'panels'> & {
//   panels?: SavedDashboardPanel[];
// };

export interface SavedDashboardPanelMap {
  [key: string]: SavedDashboardPanel;
}

export interface StagedFilter {
  field: string;
  value: string;
  operator: string;
  index: string;
}
