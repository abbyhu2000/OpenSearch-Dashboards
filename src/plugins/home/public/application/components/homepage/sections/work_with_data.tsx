/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiCard, EuiImage } from '@elastic/eui';
import { i18n } from '@osd/i18n';
import { Section } from '../../../../services/section_type/section_type';
import { renderFn } from './utils';
import { getServices } from '../../../opensearch_dashboards_services';

const render = renderFn(() => {
  const services = getServices();
  const navigateToApp = services.application.navigateToApp;
  const navigateToUrl = services.application.navigateToUrl;
  const darkMode = services.injectedMetadata.getBranding().darkMode;

  return (
    <EuiFlexGroup wrap direction="row" alignItems="stretch">
      <EuiFlexItem>
        <EuiCard
          title={i18n.translate('home.sections.workWithData.sampleData.title', {
            defaultMessage: 'Start with a sample data set',
          })}
          titleSize="xs"
          textAlign="left"
          description={i18n.translate('home.sections.workWithData.sampleData.description', {
            defaultMessage: 'with Sample Data Sets',
          })}
          onClick={() => navigateToApp('home', { path: '#/tutorial_directory' })}
          image={
            <EuiImage
              src={services.addBasePath(
                `/plugins/home/assets/sample_data_resources/ecommerce/${
                  darkMode ? 'dashboard_dark_new.png' : 'dashboard_new.png'
                }`
              )}
              alt="Sample data image"
            />
          }
        />
      </EuiFlexItem>
      <EuiFlexItem>
        <EuiCard
          title={i18n.translate('home.sections.workWithData.ingest.title', {
            defaultMessage: 'Prepare and ingest data',
          })}
          titleSize="xs"
          textAlign="left"
          description={i18n.translate('home.sections.workWithData.ingest.description', {
            defaultMessage: 'with Data Prepper',
          })}
          onClick={() => navigateToUrl('https://opensearch.org/docs/latest/data-prepper/')}
          image={
            <EuiImage
              src={services.addBasePath(`/plugins/home/assets/ingestion.png`)}
              alt="Injestion image"
            />
          }
        />
      </EuiFlexItem>
      <EuiFlexItem>
        <EuiCard
          title={i18n.translate('home.sections.workWithData.dashboards.title', {
            defaultMessage: 'Set up pre-built dashboards',
          })}
          titleSize="xs"
          textAlign="left"
          description={i18n.translate('home.sections.workWithData.integration.description', {
            defaultMessage: 'with integrations',
          })}
          onClick={() => navigateToApp('dashboards')}
        />
      </EuiFlexItem>
      <EuiFlexItem>
        <EuiCard
          title={i18n.translate('home.sections.workWithData.discover.title', {
            defaultMessage: 'Query and filter data for in-depth analysis',
          })}
          titleSize="xs"
          textAlign="left"
          description={i18n.translate('home.sections.workWithData.discover.description', {
            defaultMessage: 'with Discover',
          })}
          onClick={() => navigateToApp('discover')}
          image={
            <EuiImage
              src={services.addBasePath(
                `/plugins/home/assets/${darkMode ? 'discover_dark.png' : 'discover_light.png'}`
              )}
              alt="Explore image"
            />
          }
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
});

export const workWithDataSection: Section = {
  id: 'home:workWithData',
  title: i18n.translate('home.sections.workWithData.title', {
    defaultMessage: 'Start working with data',
  }),
  description: i18n.translate('home.sections.workWithData.description', {
    defaultMessage:
      'Get started by ingesting data to access robust and visual data exploration capabilities. Not ready to ingest data yet? Use our sample data to explore and get familiar with OpenSearch capabilities.',
  }),
  render,
};
