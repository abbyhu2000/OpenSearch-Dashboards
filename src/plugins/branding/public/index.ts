import './index.scss';

import { BrandingPlugin } from './plugin';

// This exports static code and TypeScript types,
// as well as, OpenSearch Dashboards Platform `plugin()` initializer.
export function plugin() {
  return new BrandingPlugin();
}
export { BrandingPluginSetup, BrandingPluginStart } from './types';
