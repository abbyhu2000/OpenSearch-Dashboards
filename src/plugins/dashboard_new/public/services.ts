/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  ApplicationStart,
  Capabilities,
  ChromeStart,
  HttpStart,
  I18nStart,
  IUiSettingsClient,
  OverlayStart,
  SavedObjectsStart,
} from '../../../core/public';
import { TypesStart } from './vis_types';
import { createGetterSetter } from '../../opensearch_dashboards_utils/common';
import {
  DataPublicPluginStart,
  FilterManager,
  IndexPatternsContract,
  TimefilterContract,
} from '../../../plugins/data/public';
import { UsageCollectionSetup } from '../../usage_collection/public';
import { ExpressionsStart } from '../../expressions/public';
import { UiActionsStart } from '../../ui_actions/public';
import { SavedDashboardsLoader } from './saved_dashboards';
import { SavedObjectLoader } from '../../saved_objects/public';
import { EmbeddableStart } from '../../embeddable/public';

export const [getUISettings, setUISettings] = createGetterSetter<IUiSettingsClient>('UISettings');

export const [getCapabilities, setCapabilities] = createGetterSetter<Capabilities>('Capabilities');

export const [getHttp, setHttp] = createGetterSetter<HttpStart>('Http');

export const [getApplication, setApplication] = createGetterSetter<ApplicationStart>('Application');

export const [getEmbeddable, setEmbeddable] = createGetterSetter<EmbeddableStart>('Embeddable');

export const [getSavedObjects, setSavedObjects] = createGetterSetter<SavedObjectsStart>(
  'SavedObjects'
);

export const [getTypes, setTypes] = createGetterSetter<TypesStart>('Types');

export const [getI18n, setI18n] = createGetterSetter<I18nStart>('I18n');

export const [getFilterManager, setFilterManager] = createGetterSetter<FilterManager>(
  'FilterManager'
);

export const [getTimeFilter, setTimeFilter] = createGetterSetter<TimefilterContract>('TimeFilter');

export const [getIndexPatterns, setIndexPatterns] = createGetterSetter<IndexPatternsContract>(
  'IndexPatterns'
);

export const [getSearch, setSearch] = createGetterSetter<DataPublicPluginStart['search']>('Search');

export const [getUsageCollector, setUsageCollector] = createGetterSetter<UsageCollectionSetup>(
  'UsageCollection'
);

export const [getExpressions, setExpressions] = createGetterSetter<ExpressionsStart>('Expressions');

export const [getUiActions, setUiActions] = createGetterSetter<UiActionsStart>('UiActions');

export const [getSavedDashboardsLoader, setSavedDashboardsLoader] = createGetterSetter<
  SavedDashboardsLoader
>('SavedDashboardsLoader');

export const [getAggs, setAggs] = createGetterSetter<DataPublicPluginStart['search']['aggs']>(
  'AggConfigs'
);

export const [getOverlays, setOverlays] = createGetterSetter<OverlayStart>('Overlays');


