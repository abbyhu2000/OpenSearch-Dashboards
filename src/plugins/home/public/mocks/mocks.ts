/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { applicationServiceMock, chromeServiceMock, coreMock } from '../../../../core/public/mocks';
import { HomeOpenSearchDashboardsServices } from '../application/opensearch_dashboards_services';
import { injectedMetadataServiceMock } from '../../../../core/public/mocks';
import { featureCatalogueRegistryMock } from '../services/feature_catalogue/feature_catalogue_registry.mock';
import { environmentServiceMock } from '../services/environment/environment.mock';
import { tutorialServiceMock } from '../services/tutorials/tutorial_service.mock';
import { sectionTypeServiceMock } from '../services/section_type/section_type.mock';
import { configSchema } from '../../config';

const createSetupContract = () => ({
  featureCatalogue: featureCatalogueRegistryMock.createSetup(),
  environment: environmentServiceMock.createSetup(),
  tutorials: tutorialServiceMock.createSetup(),
  sectionTypes: sectionTypeServiceMock.createSetup(),
});

export const createStartContract = () => {
  const coreStartMock = coreMock.createStart();
  const chromeMock = chromeServiceMock.createStartContract();
  const applicationMock = applicationServiceMock.createStartContract();
  const toastNotifications = coreStartMock.notifications.toasts;
  const injectedMetadataMock = injectedMetadataServiceMock.createStartContract();
  injectedMetadataMock.getBranding.mockReturnValue({});
  const addBasePathMock = jest.fn((path: string) => (path ? path : 'path'));
  const opensearchDashboardsVersion = '3.0.0';

  return ({
    ...coreStartMock,
    chrome: chromeMock,
    application: applicationMock,
    injectedMetadata: injectedMetadataMock,
    addBasePath: addBasePathMock,
    getBasePath: jest.fn(),
    toastNotifications,
    opensearchDashboardsVersion,
    trackUiMetric: jest.fn(),
    savedObjectsClient: {
      find: async () => {
        return {
          total: 0,
        };
      },
    },
  } as unknown) as jest.Mocked<HomeOpenSearchDashboardsServices>;
};

export const homePluginMock = {
  createSetupContract,
  createStartContract,
};
