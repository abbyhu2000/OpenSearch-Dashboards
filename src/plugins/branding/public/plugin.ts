import { i18n } from '@osd/i18n';
import { PluginInitializerContext } from 'opensearch-dashboards/server';
import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '../../../core/public';
import { BrandingPluginSetup, BrandingPluginStart, AppPluginStartDependencies } from './types';
import { PLUGIN_NAME } from '../common';
import { ConfigSchema } from '../config';
import { BrandingAssignment } from './services';

export class BrandingPlugin implements Plugin<BrandingPluginSetup, BrandingPluginStart> {
  private readonly brandingAssignment = new BrandingAssignment();

  constructor(private readonly initializerContext: PluginInitializerContext<ConfigSchema>) {}

  public setup(core: CoreSetup): BrandingPluginSetup {
    // Register an application into the side navigation menu
    core.application.register({
      id: 'branding',
      title: PLUGIN_NAME,
      async mount(params: AppMountParameters) {
        // Load application bundle
        const { renderApp } = await import('./application');
        // Get start services as specified in opensearch_dashboards.json
        const [coreStart, depsStart] = await core.getStartServices();
        // Render the application
        return renderApp(coreStart, depsStart as AppPluginStartDependencies, params);
      },
    });

    // Return methods that should be available to other plugins
    return {
      getGreeting() {
        return i18n.translate('branding.greetingText', {
          defaultMessage: 'Hello from {name}!',
          values: {
            name: PLUGIN_NAME,
          },
        });
      },
    };
  }

  public start(core: CoreStart): BrandingPluginStart {
    return {};
  }

  public stop() {}
}
