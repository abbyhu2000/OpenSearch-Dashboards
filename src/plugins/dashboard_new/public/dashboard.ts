/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/**
 * @name Dashboard
 *
 * @description This class consists of aggs, params, listeners, title, and type.
 *  - Aggs: Instances of IAggConfig.
 *  - Params: The settings in the Options tab.
 *
 * Not to be confused with vislib/vis.js.
 */

import { isFunction, defaults, cloneDeep } from 'lodash';
import { Assign } from '@osd/utility-types';
import { i18n } from '@osd/i18n';
import { PersistedState } from './persisted_state';
import { getTypes, getAggs, getSearch, getSavedSearchLoader } from './services';
import {
  IAggConfigs,
  IndexPattern,
  ISearchSource,
  AggConfigOptions,
  SearchSourceFields,
} from '../../../plugins/data/public';
import { SerializedDashboard } from '../dashboard';

export interface DashboardParams {
  [key: string]: any;
}

// type PartialDashboardState = Assign<SerializedVis, { data: Partial<SerializedVisData> }>;

// id?: string;
// timeRestore: boolean;
// timeTo?: string;
// timeFrom?: string;
// description?: string;
// panelsJSON: string;
// optionsJSON?: string;
// // TODO: write a migration to rid of this, it's only around for bwc.
// uiStateJSON?: string;
// lastSavedTitle: string;
// refreshInterval?: RefreshInterval;
// searchSource: ISearchSource;
// getQuery(): Query;
// getFilters(): Filter[];

export class Dashboard<TDashboardParams = DashboardParams> {
  public readonly id?: string;
  public title: string = '';
  public description: string = '';
  public params: TDashboardParams;
  // Session state is for storing information that is transitory, and will not be saved with the visualization.
  // For instance, map bounds, which depends on the view port, browser window size, etc.
  public sessionState: Record<string, any> = {};

  public readonly uiState: PersistedState;

  constructor(dashboardState: SerializedDashboard = {} as any) {
    this.params = this.getParams(dashboardState.params);
    this.uiState = new PersistedState(dashboardState.uiState);
    this.id = dashboardState.id;
  }

  private getParams(params: VisParams) {
    return defaults({}, cloneDeep(params ?? {}), cloneDeep(this.type.visConfig?.defaults ?? {}));
  }

  async setState(state: PartialDashboardState) {
    let typeChanged = false;
    if (state.type && this.type.name !== state.type) {
      // @ts-ignore
      this.type = this.getType(state.type);
      typeChanged = true;
    }
    if (state.title !== undefined) {
      this.title = state.title;
    }
    if (state.description !== undefined) {
      this.description = state.description;
    }
    if (state.params || typeChanged) {
      this.params = this.getParams(state.params);
    }
    if (state.data && state.data.searchSource) {
      this.data.searchSource = await getSearch().searchSource.create(state.data.searchSource!);
      this.data.indexPattern = this.data.searchSource.getField('index');
    }
    if (state.data && state.data.savedSearchId) {
      this.data.savedSearchId = state.data.savedSearchId;
      if (this.data.searchSource) {
        this.data.searchSource = await getSearchSource(
          this.data.searchSource,
          this.data.savedSearchId
        );
        this.data.indexPattern = this.data.searchSource.getField('index');
      }
    }
    if (state.data && (state.data.aggs || !this.data.aggs)) {
      const aggs = state.data.aggs ? cloneDeep(state.data.aggs) : [];
      const configStates = this.initializeDefaultsFromSchemas(aggs, this.type.schemas.all || []);
      if (!this.data.indexPattern) {
        if (aggs.length) {
          const errorMessage = i18n.translate(
            'visualizations.initializeWithoutIndexPatternErrorMessage',
            {
              defaultMessage: 'Trying to initialize aggs without index pattern',
            }
          );
          throw new Error(errorMessage);
        }
        return;
      }
      this.data.aggs = getAggs().createAggConfigs(this.data.indexPattern, configStates);
    }
  }

  clone() {
    const { data, ...restOfSerialized } = this.serialize();
    const vis = new Vis(this.type.name, restOfSerialized as any);
    vis.setState({ ...restOfSerialized, data: {} });
    const aggs = this.data.indexPattern
      ? getAggs().createAggConfigs(this.data.indexPattern, data.aggs)
      : undefined;
    vis.data = {
      ...this.data,
      aggs,
    };
    return vis;
  }

  serialize(): SerializedVis {
    const aggs = this.data.aggs ? this.data.aggs.aggs.map((agg) => agg.toJSON()) : [];
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      type: this.type.name,
      params: cloneDeep(this.params) as any,
      uiState: this.uiState.toJSON(),
      data: {
        aggs: aggs as any,
        searchSource: this.data.searchSource ? this.data.searchSource.getSerializedFields() : {},
        savedSearchId: this.data.savedSearchId,
      },
    };
  }

  // deprecated
  isHierarchical() {
    if (isFunction(this.type.hierarchicalData)) {
      return !!this.type.hierarchicalData(this);
    } else {
      return !!this.type.hierarchicalData;
    }
  }

  private initializeDefaultsFromSchemas(configStates: AggConfigOptions[], schemas: any) {
    // Set the defaults for any schema which has them. If the defaults
    // for some reason has more then the max only set the max number
    // of defaults (not sure why a someone define more...
    // but whatever). Also if a schema.name is already set then don't
    // set anything.
    const newConfigs = [...configStates];
    schemas
      .filter((schema: any) => Array.isArray(schema.defaults) && schema.defaults.length > 0)
      .filter(
        (schema: any) => !configStates.find((agg) => agg.schema && agg.schema === schema.name)
      )
      .forEach((schema: any) => {
        const defaultSchemaConfig = schema.defaults.slice(0, schema.max);
        defaultSchemaConfig.forEach((d: any) => newConfigs.push(d));
      });
    return newConfigs;
  }
}
