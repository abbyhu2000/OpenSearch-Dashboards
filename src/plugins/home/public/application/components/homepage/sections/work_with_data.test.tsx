/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { workWithDataSection } from './work_with_data';
import {
  HomeOpenSearchDashboardsServices,
  setServices,
} from '../../../opensearch_dashboards_services';

import { BehaviorSubject } from 'rxjs';
import { PublicAppInfo } from 'opensearch-dashboards/public';
import { homePluginMock } from 'src/plugins/home/public/mocks/mocks';

let mockElement: HTMLElement;
let services: HomeOpenSearchDashboardsServices;

describe('Work with data section', () => {
  beforeAll(() => {
    mockElement = document.createElement('element');
    services = homePluginMock.createStartContract();
  });

  it('renders without observability card', async () => {
    setServices(services);
    workWithDataSection.render(mockElement);
    expect(mockElement).toMatchSnapshot();
  });

  it('renders with observability card named Set up pre-built dashboards', async () => {
    const app1: PublicAppInfo = {
      id: 'id',
      title: 'title',
      status: 0,
      navLinkStatus: 0,
      appRoute: 'string',
    };
    const app2: PublicAppInfo = {
      id: 'observability-dashboards',
      title: 'observability-dashboards',
      status: 0,
      navLinkStatus: 0,
      appRoute: 'string',
    };
    const appMap: ReadonlyMap<string, PublicAppInfo> = new Map<string, PublicAppInfo>([
      ['app1', app1],
      ['app2', app2],
    ]);
    services.application.applications$ = new BehaviorSubject<ReadonlyMap<string, PublicAppInfo>>(
      appMap
    );

    setServices(services);
    workWithDataSection.render(mockElement);
    expect(mockElement).toMatchSnapshot();
  });
});
