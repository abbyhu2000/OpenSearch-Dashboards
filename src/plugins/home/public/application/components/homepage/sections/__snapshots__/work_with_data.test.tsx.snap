// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work with data section renders with observability card named Set up pre-built dashboards 1`] = `
<element>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
  >
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Sample data image"
                class="euiImage__img"
                src="/plugins/home/assets/sample_data_resources/ecommerce/dashboard_new.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              Start with a sample data set
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Sample Data Sets
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Injestion image"
                class="euiImage__img"
                src="/plugins/home/assets/ingestion.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              <p>
                Prepare and ingest data
                <span
                  class="popover-title-icon"
                  data-euiicon-type="popout"
                />
              </p>
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Data Prepper
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Integration image"
                class="euiImage__img"
                src="/plugins/home/assets/integration.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              Set up pre-built dashboards
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with integrations
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Explore image"
                class="euiImage__img"
                src="/plugins/home/assets/discover_light.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              Query and filter data for in-depth analysis
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Discover
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</element>
`;

exports[`Work with data section renders without observability card 1`] = `
<element>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
  >
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Sample data image"
                class="euiImage__img"
                src="/plugins/home/assets/sample_data_resources/ecommerce/dashboard_new.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              Start with a sample data set
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Sample Data Sets
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Injestion image"
                class="euiImage__img"
                src="/plugins/home/assets/ingestion.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              <p>
                Prepare and ingest data
                <span
                  class="popover-title-icon"
                  data-euiicon-type="popout"
                />
              </p>
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Data Prepper
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem"
    >
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--leftAligned euiCard--isClickable"
      >
        <div
          class="euiCard__top"
        >
          <div
            class="euiCard__image"
          >
            <figure
              class="euiImage euiImage--original"
            >
              <img
                alt="Explore image"
                class="euiImage__img"
                src="/plugins/home/assets/discover_light.png"
              />
            </figure>
          </div>
        </div>
        <div
          class="euiCard__content"
        >
          <span
            class="euiTitle euiTitle--xsmall euiCard__title"
            id="generated-idTitle"
          >
            <button
              aria-describedby=" generated-idDescription"
              class="euiCard__titleButton"
            >
              Query and filter data for in-depth analysis
            </button>
          </span>
          <div
            class="euiText euiText--small euiCard__description"
            id="generated-idDescription"
          >
            <p>
              with Discover
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</element>
`;
